xds/src/main/java/io/grpc/xds/LocalityStore.java

```diff
@@ -306,13 +307,9 @@ private void deactivate(final XdsLocality locality) {
         public void run() {
           localityLbInfo.shutdown();
 
-          ImmutableMap.Builder<XdsLocality, LocalityLbInfo> builder = ImmutableMap.builder();
-          for (Map.Entry<XdsLocality, LocalityLbInfo> entry : localityMap.entrySet()) {
-            if (!entry.getKey().equals(locality)) {
-              builder.put(entry);
-            }
-          }
-          localityMap = builder.build();
+          Map<XdsLocality, LocalityLbInfo> copy = new LinkedHashMap<>(localityMap);
```

Comment:
You can just make `localityMap` itself a `LinkedHashMap`, as we discussed in https://github.com/grpc/grpc-java/pull/6110#discussion_r321533255. It doesn't have to be immutable.

Reply:
PTALA